impl/spanner/testutil/testutil.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package testutil helps with tests against Spanner.
+package testutil
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/spanner"
+	"cloud.google.com/go/spanner/spannertest"
+	"google.golang.org/api/option"
+	"google.golang.org/grpc"
+
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	databasepb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+)
+
+var dbCount struct {
+	sync.Mutex
+	count int
+}
+
+func uniqueDBName() string {
+	dbCount.Lock()
+	database := fmt.Sprintf("fake-db%d", dbCount.count)
+	dbCount.count++
+	dbCount.Unlock()
+	const instance = "fake-instance"
+	const project = "fake-proj"
+	return fmt.Sprintf("projects/%s/instances/%s/databases/%s", project, instance, database)
+}
+
+func CreateDatabase(ctx context.Context, t testing.TB, ddlStatements []string) (*spanner.Client, func()) {
```

Comment:
Please add comments for the exported stuff.

Reply:
done